assets/src/edit-story/components/dropDown/index.js

```diff
@@ -0,0 +1,314 @@
+  margin: 2px 0 0;
+  font-size: 14px;
+  text-align: left;
+  list-style: none;
+  background-color: #fff;
+  background-clip: padding-box;
+  border: 1px solid #ccc;
+  border: 1px solid rgba(0, 0, 0, 0.15);
+  border-radius: 4px;
+  box-shadow: 0 6px 12px rgba(0, 0, 0, 0.175);
+
+  li {
+    cursor: pointer;
+    :hover {
+      background: #ddd;
+    }
+    & > a {
+      display: block;
+      padding: 5px 20px;
+      clear: both;
+      font-weight: 400;
+      line-height: 1.42857143;
+      color: #333;
+      white-space: nowrap;
+      text-decoration: none;
+    }
+  }
+`;
+
+const DropDownItem = styled.li.attrs({ tabIndex: '0', role: 'option' })`
+  font-size: ${({ theme }) => theme.fonts.body2.size};
+  line-height: ${({ theme }) => theme.fonts.body2.lineHeight};
+  letter-spacing: ${({ theme }) => theme.fonts.body2.letterSpacing};
+  padding: 16px;
+  margin: 0;
+
+  &:hover {
+    background-color: ${({ theme }) => theme.colors.bg.v11};
+  }
+
+  &:focus {
+    background-color: ${({ theme }) => theme.colors.bg.v11};
+    outline: none;
+  }
+`;
+
+const DropDownItemSpan = styled.a``;
+
+function DropDown({ options, value, onChange, disabled }) {
+  DropDown.selectRef = React.createRef();
+  DropDown.arrayOfOptionsRefs = [];
+
+  const [openOptions, setOpenOptions] = useState(false);
+  const [focusedOption, setFocusedOption] = useState(undefined);
+  const [searchValue, setSearchValue] = useState('');
+  const [activeItem, setActiveItem] = useState(
+    options.find((item) => item.value === value)
+  );
+
+  useEffect(() => {
+    setActiveItem(options.find((item) => item.value === value));
+  }, [value, options]);
+
+  DropDown.handleClickOutside = () => setOpenOptions(false);
+
+  const clearOptionsRefs = () => {
+    DropDown.arrayOfOptionsRefs = [];
+  };
+
+  const handleOpenOptions = (event) => {
+    switch (event.type) {
+      case 'click':
+        setOpenOptions(!openOptions);
+        setFocusedOption(document.activeElement.id);
+        break;
+      case 'keydown':
+        if (event.key === 'Enter' || event.key === ' ') {
+          _handleOpenOptions(event);
+        }
+        break;
+      default:
+    }
+  };
+
+  const _handleOpenOptions = useCallback(() => {
+    setOpenOptions(!openOptions);
+  }, [openOptions]);
+
+  const handleCurrentValue = useCallback(
+    (option) => {
+      if (onChange) {
+        onChange(option);
+      }
+    },
+    [onChange]
+  );
+
+  const handleSelectElement = (event, index) => {
+    event.preventDefault();
+    if (index >= 0 && index < DropDown.arrayOfOptionsRefs.length) {
+      DropDown.arrayOfOptionsRefs[index].focus();
+      setFocusedOption(document.activeElement.id);
+    }
+  };
+
+  const handleOptionsEvents = (option, index, event) => {
+    switch (event.type) {
+      case 'click':
+        handleCurrentValue(option);
+        setOpenOptions(!openOptions);
+        break;
+      case 'keydown':
+        if (event.key === 'Enter' || event.key === ' ') {
```

Comment:
Wouldn't it be better to have the swtich statement all the list of key events. There are only two types of events...

Reply:
So currently we have two types `click` and `keydown` and keydown has several options to separate the features including normal keydown. What would you suggest? Using switch for event.key?

Reply:
Can you refactor this to use `useKeyDownEffect` instead for all keydown events?

`handleOptionsEvents` would then turn into a `handleClick` callback

Reply:
Okay

Reply:
All the key events should be handled by useKeyDownEffect - it'll create a much nicer interface and make the code a lot leaner and cleaner.

Reply:
Refactored to use the useKeyDownEffect for all events. But some kind of static definition of a to z and 1 to 9.